cranelift-entity/src/set.rs

```diff
@@ -93,6 +103,17 @@ where
         self.elems[index / 8] |= 1 << (index % 8);
         result
     }
+
+    /// Removes and returns the last `K` from the set if any are present.
+    pub fn pop(&mut self) -> Option<K> {
+        if let Some(k) = self.keys().next_back() {
```

Comment:
Tgis doesnt give the last set key, but the key at `self.len * 8`, which may have already been removed.

Reply:
How would you suggest to get the last key?

Reply:
Search last item of `self.elems` for last set bit. If the resulting u8 is zero, pop it. Not sure if it is correct. In any case you will need to decrement `self.len` afterwards.

Reply:
@bjorn3 Okay, I've reimplemented and added a test to make sure it works.